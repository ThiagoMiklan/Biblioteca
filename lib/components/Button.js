"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var React = _interopRequireWildcard(require("react"));

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  disabled: 'boolean',
  "delete": 'boolean',
  definition: 'string',
  custom: 'object',
  onClick: 'function'
};

var Button = function Button(props) {
  (0, _type_validations.validate)(props, props_obj, "Button");
  var definition = getDefinition(props);
  return /*#__PURE__*/React.createElement("button", (0, _extends2["default"])({}, props.custom, {
    disabled: props.disabled === true ? true : false,
    className: definition,
    onClick: props.onClick
  }), props.label);
};

function getDefinition(props) {
  var definition = "";

  if (props["delete"] === true) {
    definition = "delete";
  } else {
    definition = props.definition === undefined ? "button" : "button " + props.definition;
  }

  return definition;
}

var _default = Button;
exports["default"] = _default;