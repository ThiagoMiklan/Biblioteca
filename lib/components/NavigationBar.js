"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _typeof2 = _interopRequireDefault(require("@babel/runtime/helpers/typeof"));

var _react = _interopRequireDefault(require("react"));

var _type_validations = require("../tools/type_validations");

var props_obj = {
  definition: "string",
  link: "string",
  src: "string",
  width_brand: "number",
  alt: "string",
  height_brand: "number",
  itens_start: "array",
  itens_end: "array"
};

var NavigationBar = function NavigationBar(props) {
  (0, _type_validations.validate)(props, props_obj, "NavigationBar");
  var definition = props.definition == undefined ? "navbar" : "navbar " + props.definition;
  return /*#__PURE__*/_react.default.createElement("nav", {
    className: definition
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "navbar-brand"
  }, /*#__PURE__*/_react.default.createElement("a", {
    className: "navbar-item",
    href: props.link
  }, /*#__PURE__*/_react.default.createElement("img", {
    src: props.src,
    alt: props.alt,
    width: props.width_brand,
    height: props.height_brand
  }))), /*#__PURE__*/_react.default.createElement("div", {
    className: "navbar-menu"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "navbar-start"
  }, assemble(props.itens_start)), /*#__PURE__*/_react.default.createElement("div", {
    className: "navbar-end"
  }, assemble(props.itens_end))));
};

function assemble(itens) {
  var itens_code = ""; // verifica se array

  if (Array.isArray(itens)) {
    // elimina vazio
    if (itens.length > 0) {
      // se não estiver vazio monta uma lista de itens
      itens_code = assembleItens(itens);
    }
  } else if ((0, _typeof2.default)(itens) == 'object' && itens != undefined) {
    //se não for array, não for undefined e for objeto realiza operação
    itens_code = assembleItem(itens);
  }

  return itens_code;
}

function assembleItens(itens) {
  var size = itens.length;
  var itens_code = itens.map(function (item) {
    return decideTypeOfItem(item);
  });
  return itens_code;
}

function decideTypeOfItem(item) {
  var itens_code = ""; // se um item for um array (array dentro do array), então ele é um dropdown

  if (Array.isArray(item)) {
    itens_code = assembleItensDropDown(item);
  } else {
    // se não ele conjunto de itens encadeados
    itens_code = assembleItem(item);
  }

  return itens_code;
} // default : hoverable


function assembleItensDropDown(itens) {
  var firstitemDropDown = assembleFirstItemDropDown(itens[0]);

  if (itens.length > 0) {
    itens.splice(0, 1);
  }

  var itens_code = /*#__PURE__*/_react.default.createElement("div", {
    className: "navbar-item has-dropdown is-hoverable"
  }, firstitemDropDown, /*#__PURE__*/_react.default.createElement("div", {
    className: "navbar-dropdown"
  }, itens.map(function (item) {
    return assembleItem(item);
  })));

  return itens_code;
}

function assembleFirstItemDropDown(item) {
  var first_item = "";

  if (item != undefined) {
    first_item = /*#__PURE__*/_react.default.createElement("a", (0, _extends2.default)({
      className: "navbar-link",
      onClick: item["onClick"]
    }, item["custom"]), item["value"]);
  }

  return first_item;
}

function assembleItem(item) {
  var definition = item["definition"] == undefined ? "navbar-item " : "navbar-item " + item["definition"];

  var item_code = /*#__PURE__*/_react.default.createElement("a", (0, _extends2.default)({
    onClick: item["onClick"],
    className: definition
  }, item["custom"]), item["value"]);

  return item_code;
}

var _default = NavigationBar;
exports.default = _default;