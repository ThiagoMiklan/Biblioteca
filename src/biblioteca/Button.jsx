// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import {renderToString} from 'react-dom/server';

type Props = {
  disabled?: bool,
  delete?: bool,
  definition?: string,
  children?: React.Node,
  custom?: any,
  onClick?: () => void
}

/*

Possui children
Eventos: onClick no <button> </button>
Observações: 

    1 - Caso exista a necessidade de adicionar um ícone ao componente Button, é necessário seguir
             a seguinte estrutura:

            <Button definition="info" >
                <Icon icon_name="fas fa-check" [right={true} OR empty prop right]>
                    Nome do botão
                </Icon>
            </Button >

            Sendo necessário informar o componente Icon como filho do componente Button
            e o nome do botão deve ser informado no Icon e não no Button como em outras
            situações onde usa-se o componente isolado.
*/

const Button= (props: Props)=>{
    var definition = getDefinition(props);
    
     return <button 
                {...props.custom} 
                disabled = {props.disabled == true? true: false } 
                className={definition} 
                onClick={props.onClick}
                >
                {props.children}
         </button>
         ;
}

function getDefinition(props){
    var definition = "";

    if(props.delete ==  true){
        definition = "delete";
    }else{
        definition = (props.definition == undefined) ? "button" : "button "+ props.definition;
    }

    return definition;
}

Button.propTypes = {
    definition: PropTypes.string,
    onClick: PropTypes.func,
    delete: PropTypes.bool,
    disabled: PropTypes.bool
};


export default Button;