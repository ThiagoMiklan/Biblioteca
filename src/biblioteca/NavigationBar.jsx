// @flow
import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import getClassName from '../tools/getClassName';
import checkerDefinition from '../tools/checker.js';
/*
Existem algumas maneira de fornecer os itens de "start" e "end" ao usar o presente componente 
1 - Fornecer através da props itens_start ou itens_end um objeto contendo um atributo "value"
2 - Fornecer um array de objetos contendo em cada objeto o atributo value
3 - Fornecer um array de objetos com um dropdown, nesse caso o um array de objetos dentro do array em que está contido
4 - É possível fornecer componentes diversos, como Botões no start ou end da barra, para tal faz-se necessário
    informar o código do elemento.
    Exemplo:

    var itens_end = [{value: <Button>Login</Button>}, {value: <Button>Log Out</Button>}] 

    <NavigationBar
            itens_end={itens_start_3}
        />
*/
type Props = {
    definition?:string,
    link_brand?:string,
    src_brand?:string,
    width_brand?:string,
    alt?:string,
    height_brand?:string,
    itens_start?:Array<Object>,
    itens_end?:Array<Object>
}

const NavigationBar = (props:Props)=>{
    checkerDefinition(props.definition,"NavigationBar");
    var definition="navbar " + classnames(getClassName(props.definition,"NavigationBar"));
        return (
            <nav className={definition}>
                
                <div className="navbar-brand">
                    <a className="navbar-item" href={props.link_brand}>
                        <img src={props.src_brand} alt={props.alt} width={props.width_brand} height={props.height_brand}></img>
                    </a>
                </div>
                <div className="navbar-menu">
                    <div className="navbar-start">
                        {assemble(props.itens_start)}
                    </div>
                    <div className="navbar-end">
                        {assemble(props.itens_end)}
                    </div>
                </div>
            </nav>

        );
    
}

function assemble(itens) {
    var itens_code = "";
    // verifica se array
    if (Array.isArray(itens)) {
        // elimina vazio
        if(itens.length > 0){
            // se não estiver vazio monta uma lista de itens
            itens_code = assembleItens(itens);
        }
    } else  if(typeof itens =='object'  && itens != undefined){
        //se não for array, não for undefined e for objeto realiza operação
        itens_code = assembleItem(itens, itens["definition"]);
    }
    return itens_code;
}

function assembleItens(itens) {
    let size = itens.length;
    var itens_code  = itens.map(item => decideTypeOfItem(item));
    return itens_code;
}


function decideTypeOfItem(item){
    var itens_code ="";
     // se um item for um array (array dentro do array), então ele é um dropdown
    if(Array.isArray(item)){
        itens_code = assembleItensDropDown(item);
    }else{
    // se não ele conjunto de itens encadeados
        itens_code = assembleItem(item, "");
    }
    return itens_code;
}

// default : hoverable
function assembleItensDropDown(itens){
    var firstitemDropDown = assembleFirstItemDropDown(itens[0]);
   
    if(itens.length > 0){
        itens.splice(0,1);
    }
    var itens_code =
    <div className="navbar-item has-dropdown is-hoverable">
        {firstitemDropDown}
        <div className="navbar-dropdown">
            {itens.map(item => assembleItem(item, ""))}
        </div>
    </div>;

    

    return itens_code;
}

function assembleFirstItemDropDown(item) {
    var first_item = "";
    if (item != undefined) {
        first_item = <a className="navbar-link">
            {item["value"]}
        </a>
    }

   
    return first_item;
}

function assembleItem(item, definition) {
    var className = "navbar-item "+ classnames(getClassName(item["definition"], "NavigationBar"));
    var x = item["value"];
    var item_code = <a onClick={item["onClick"]} className={className}>{item["value"]}</a>;
  
    return item_code;
}

NavigationBar.propTypes = {
    src_brand: PropTypes.string,
    link_brand: PropTypes.string,
    width_brand: PropTypes.number,
    height_brand: PropTypes.number,
    alt_brand: PropTypes.string,
    itens_start : PropTypes.array,
    itens_end : PropTypes.array,
    definition: PropTypes.string
}

export default NavigationBar;

