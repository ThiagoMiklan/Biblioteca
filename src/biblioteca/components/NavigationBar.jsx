import React from 'react';
import PropTypes from 'prop-types';
import classname from 'classnames';
import getClassName from '../../tools/getClassName';
import { renderToString } from 'react-dom/server';

class NavigationBar extends React.Component {



    render() {
       // var startItens = getStartItens(this.props.itens);
        //var endItens  =  getEndItens(this.props.itens);
        return (
            <nav class="navbar">
                <div class="navbar-brand">
                    <a class="navbar-item" href={this.props.link_brand}>
                        <img src={this.props.src_brand} alt={this.props.alt} width={this.props.width_brand} height={this.props.height_brand}></img>
                    </a>
                </div>
                <div className="navbar-menu">
                    <div className="navbar-start">
                        {assemble(this.props.itens_start)}
                    </div>
                    <div className="navbar-end">
                        {assemble(this.props.itens_end)}
                    </div>
                </div>
            </nav>

        );
    }

}
// Método passado para função filter da propriedade itens para que filter os itens que irão no start da barra
function filterStartItens(item) {
    var start_condition = (item["start"] == true && (item["end"] == false || item["end"] == undefined)) ;
    // default condition para casa não seja informado nem start e nem end, acontecendo isso é inserido no start
    var default_condition = (item["start"] == undefined || item["end"] == undefined);

    return start_condition || default_condition; 
}

//Método passado para função filter da propriedade itens para que filter os itens que irão no start da barra
function filterEndItens(item) {
    return item["end"] == true && (item["start"] == false || item["start"] == undefined);
}

function getStartItens(itens) {
    return itens.filter(filterStartItens);
}

function getEndItens(itens) {
    return itens.filter(filterEndItens);
}

function assemble(itens) {
    var itens_code = "";
    
    if (Array.isArray(itens)) {
        // no caso é necessário adicionar primeiro um item para que o dropdown possa funcionar
        if(itens.length > 0){
            itens_code = assembleItens(itens);
        }
    } else {
        itens_code = assembleItem(itens, itens["definition"]);
    }

    var itens_code_string = renderToString(itens_code);
    return itens_code;
}

function assembleItens(itens) {

    var itens_code =
        <div className="navbar-item has-dropdown is-hoverable">
            <a className="navbar-link">
                More
            </a>
            <div className="navbar-dropdown">
                {itens.map(item => assembleItem(item, ""))}
        </div>
        </div>;
    return itens_code;

}

function assembleItem(item, definition) {
    var className = "navbar-item " + classname(getClassName(definition, "NavigationBar"));
    var item_code = <a className={className}>{item["value"]}</a>;
    
    return item_code;
}

NavigationBar.propTypes = {
    src_brand: PropTypes.string,
    link_brand: PropTypes.string,
    width_brand: PropTypes.number,
    height_brand: PropTypes.number,
    alt_brand: PropTypes.string,
    itens_start : PropTypes.array,
    itens_end : PropTypes.array
}

export default NavigationBar;

