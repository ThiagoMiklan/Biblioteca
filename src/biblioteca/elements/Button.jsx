import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import button_props from '../../props/ButtonProps.js'
import getClassNames from '../../tools/getClassName';

/*
Observações: 

    1 - Caso exista a necessidade de adicionar um ícone ao componente Button, é necessário seguir
             a seguinte estrutura:

            <Button definition="info" >
                <Icon icon_name="fas fa-check" [right={true} OR empty prop right]>
                    Nome do botão
                </Icon>
            </Button >

            Sendo necessário informar o componente Icon como filho do componente Button
            e o nome do botão deve ser informado no Icon e não no Button como em outras
            situações onde usa-se o componente isolado.
*/


class Button extends React.Component {

    constructor(props){
        super(props);
        this.state  = {definition:props.definition};
    }

    render() {
        var definition = getDefinition(this.props.delete, this.state.definition);

        return (
            <button disabled = {this.props.disabled == true? true: false } className={definition} onClick={getClickEvent(this.props.onClick).bind(this)}>
                {this.props.children}
         </button>
         );
    }
}

/* Se propriedade isDelete é verdadeira, então
    retorna somente o className "definition", não
    envolvendo outras classes do Bulma.
*/
function getDefinition(isDelete,definition){
    if(isDelete){
       return "delete";
    }else{
        return classnames(getClassNames("button "+ definition,"Button"));
    }
}
// caso seja informado onClick ele retorna a função retornada, senão retorna uma função vazia. Método criado para evitar problemas com undefined
function getClickEvent(onClick){

    if(typeof onClick == "function"){
        return onClick;
    }else{
        return ()=> {};
    }
    
}



Button.propTypes = {
    // definição CSS Bulma
    definition: PropTypes.string,
    // Método Onclick disponível
    onClick: PropTypes.func,
    // Caso exista um ícone, informar através da prop icon, passando o componente Icon
    icon: PropTypes.element,
    // Caso o botão seja do tipo delete
    delete: PropTypes.bool
};


export default Button;