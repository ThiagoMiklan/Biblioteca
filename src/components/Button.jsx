// @flow
import * as React from 'react';
import { validate } from '../tools/type_validations.js';

// custom

type Props = {
    disabled?: bool,
    delete?: bool,
    label?: React.Node,
    definition?: string,
    children?: React.Node,
    custom?: Object,
    onClick?: () => void
}

const props_obj = {
    disabled: 'boolean',
    delete: 'boolean',
    definition: 'string',
    custom: 'object',
    onClick: 'function'
}

const Button = (props: Props) => {
    validate(props, props_obj, "Button");
    var definition = getDefinition(props);
    return <button
        {...props.custom}
        disabled={props.disabled === true ? true : false}
        className={definition}
        onClick={props.onClick}
    >
        {props.label}
    </button>
        ;
}

function getDefinition(props) {
    var definition = "";

    if (props.delete === true) {
        definition = "delete";
    } else {
        definition = (props.definition === undefined) ? "button" : "button " + props.definition;
    }

    return definition;
}



export default Button;