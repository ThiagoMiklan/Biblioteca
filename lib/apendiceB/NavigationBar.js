/*
import React from 'react';
import ReactDOM from 'react-dom';
import "bulma/css/bulma.css";
import Button from './components/Button.jsx';
import NavigationBar from './components/NavigationBar.jsx';
import {renderToString} from 'react-dom/server';


const itens_start = [
  { "value": "Home" },
  { "value": "Documentation" },
  [{ value: "More", definition: "is-active" },
  { value: "About", definition: "is-active", onClick: () => { alert("Clicked") } },
  { value: "Jobs" },
  { value: "Contact" },
  { value: "Report a issue" }
  ]
];

var itens_end = [{ value: <Button definition="is-info" label="Log in"/> },
{ value: <Button definition="is-primary" label="Log out"/> }
];



ReactDOM.render(
  <div className="container is-fluid">

   <NavigationBar 
      definition="primary"
      src="https://versions.bulma.io/0.7.2/images/bulma-logo.png"
      link_brand="https://bulma.io"
      width_brand={112}
      height_brand={28}
      alt_brand="Bulma: a modern CSS framework based on Flexbox"
      itens_start={itens_start}
      itens_end={itens_end}
    />
 </div>
  ,
  document.getElementById('root')
);
*/

/*
import React from 'react';
import ReactDOM from 'react-dom';
import "bulma/css/bulma.css";
import Button from './components/Button.jsx';
import NavigationBar from './components/NavigationBar.jsx';
import {renderToString} from 'react-dom/server';


const itens_start = [
  { "value": "Home" },
  { "value": "Documentation" },
  [{ value: "More", definition: "is-active" },
  { value: "About", definition: "is-active", onClick: () => { alert("Clicked") } },
  { value: "Jobs" },
  { value: "Contact" },
  { value: "Report a issue" }
  ]
];

var itens_end = [{ value: <Button definition="is-info" label="Log in"/> },
{ value: <Button definition="is-primary" label="Log out"/> }
];



ReactDOM.render(
 <nav class="navbar primary" data-reactroot="">
	<div class="navbar-brand">
		<a class="navbar-item">
			<img src="https://versions.bulma.io/0.7.2/images/bulma-logo.png" width="112" height="28"/>
		</a>
	</div>
	<div class="navbar-menu">
		<div class="navbar-start">
			<a class="navbar-item undefined">Home</a>
			<a class="navbar-item undefined">Documentation</a>
			<div class="navbar-item has-dropdown is-hoverable">
				<a class="navbar-link">More</a>
				<div class="navbar-dropdown">
					<a class="navbar-item is-active">About</a>
					<a class="navbar-item undefined">Jobs</a>
					<a class="navbar-item undefined">Contact</a>
					<a class="navbar-item undefined">Report a issue</a>
				</div>
			</div>
		</div>
		<div class="navbar-end">
			<a class="navbar-item undefined">
				<button class="button is-info">Log in</button>
			</a>
			<a class="navbar-item undefined">
				<button class="button is-primary">Log out</button>
			</a>
		</div>
	</div>
</nav>
  ,
  document.getElementById('root')
);


*/
"use strict";