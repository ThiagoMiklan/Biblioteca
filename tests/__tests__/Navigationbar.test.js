import React from 'react';
import renderer from 'react-test-renderer';
import NavigationBar from '../../src/components/NavigationBar';
import Button from '../../src/components/Button';


const itens_start = [
  { "value": "Home" },
  { "value": "Documentation" },
  [{ value: "More",definition:"is-active"}, 
  { value: "About" , definition:"is-active", onClick : ()=>{alert("Clicked") }},
  {value: "Jobs"},
  {value:"Contact"},
  {value:"Report a issue"}
  ]
];

const itens_end = [{ value: <Button definition="is-info">Sign up</Button> },
{ value: <Button definition="is-primary">Sign out</Button> }
];


test('navigationbar normal',()=>{
    const component = renderer.create(
       <NavigationBar src="https://versions.bulma.io/0.7.2/images/bulma-logo.png"
        link_brand="https://bulma.io"
        width_brand={112}
        height_brand={28}
        alt_brand="Bulma: a modern CSS framework based on Flexbox"
        itens_start={itens_start}
        itens_end={itens_end} 
      />
  ).toJSON();
    expect(component).toMatchSnapshot();
});

test('navigationbar primary',()=>{
    const component = renderer.create(
      <NavigationBar 
      definition="is-primary"
      src="https://versions.bulma.io/0.7.2/images/bulma-logo.png"
      link_brand="https://bulma.io"
      width_brand={112}
      height_brand={28}
      alt_brand="Bulma: a modern CSS framework based on Flexbox"
      itens_start={itens_start}
      itens_end={itens_end}
    />
  ).toJSON();
    expect(component).toMatchSnapshot();
});



