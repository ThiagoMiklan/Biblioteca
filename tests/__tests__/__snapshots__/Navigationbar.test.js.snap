// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigationbar normal 1`] = `
<nav
  className="navbar"
>
  <div
    className="navbar-brand"
  >
    <a
      className="navbar-item"
    >
      <img
        height={28}
        src="https://versions.bulma.io/0.7.2/images/bulma-logo.png"
        width={112}
      />
    </a>
  </div>
  <div
    className="navbar-menu"
  >
    <div
      className="navbar-start"
    >
      <a
        className="navbar-item "
      >
        Home
      </a>
      <a
        className="navbar-item "
      >
        Documentation
      </a>
      <div
        className="navbar-item has-dropdown is-hoverable"
      >
        <a
          className="navbar-link"
        >
          More
        </a>
        <div
          className="navbar-dropdown"
        >
          <a
            className="navbar-item is-active"
            onClick={[Function]}
          >
            About
          </a>
          <a
            className="navbar-item "
          >
            Jobs
          </a>
          <a
            className="navbar-item "
          >
            Contact
          </a>
          <a
            className="navbar-item "
          >
            Report a issue
          </a>
        </div>
      </div>
    </div>
    <div
      className="navbar-end"
    >
      <a
        className="navbar-item "
      >
        <button
          className="button is-info"
          disabled={false}
        />
      </a>
      <a
        className="navbar-item "
      >
        <button
          className="button is-primary"
          disabled={false}
        />
      </a>
    </div>
  </div>
</nav>
`;

exports[`navigationbar primary 1`] = `
<nav
  className="navbar is-primary"
>
  <div
    className="navbar-brand"
  >
    <a
      className="navbar-item"
    >
      <img
        height={28}
        src="https://versions.bulma.io/0.7.2/images/bulma-logo.png"
        width={112}
      />
    </a>
  </div>
  <div
    className="navbar-menu"
  >
    <div
      className="navbar-start"
    >
      <a
        className="navbar-item "
      >
        Home
      </a>
      <a
        className="navbar-item "
      >
        Documentation
      </a>
      <div
        className="navbar-item has-dropdown is-hoverable"
      >
        <a
          className="navbar-link"
          onClick={[Function]}
        >
          About
        </a>
        <div
          className="navbar-dropdown"
        >
          <a
            className="navbar-item "
          >
            Jobs
          </a>
          <a
            className="navbar-item "
          >
            Contact
          </a>
          <a
            className="navbar-item "
          >
            Report a issue
          </a>
        </div>
      </div>
    </div>
    <div
      className="navbar-end"
    >
      <a
        className="navbar-item "
      >
        <button
          className="button is-info"
          disabled={false}
        />
      </a>
      <a
        className="navbar-item "
      >
        <button
          className="button is-primary"
          disabled={false}
        />
      </a>
    </div>
  </div>
</nav>
`;
